[WIP] add blogpost about Makie showcase projects

SimonDanisch opened the pull request.

SimonDanisch commented:
Hi everyone,  

Sorry to tag you all, but I’d love to regularly feature cool projects using Makie! 😊  
This blog post is a very random selection of recent projects which caught my eye — going forward, we might set up a discord/slack/discourse channel to find more projects (I'll add details later to the post).  

**If possible**, it would be amazing if you can add a brief description or further examples to your projects section. 
Keep it short to keep the post concise, but if you’d like to share more, please contact me and we can also talk about publishing an in-depth post about the project!  

## Please answer at least these questions:
  
- [ ] Exclude my project from the blog post.
- [ ] I’ll contribute something within 1-2 weeks.  
- [ ] I’ll contribute but need more than 2 weeks.  
- [ ] I don’t have time at all, but you can write a summary for me.  
- [ ] Ping **[name]** to respond on my behalf.  

Feel free to mute this thread—just drop a quick comment if you do, so I know who will stop replying!  

Thank you all a lot for your great work! :)   

Link to the current preview:  
https://makie.org/previews/PR21/blogposts/showcases/


@Larbino1 @sally-00 
@kbarros 
@baggepinnen
@xavierr @moyner 
@magnuask 
@dgleich 
@AntonOresten 
@RGerzaguet 
@fverdugo 
@myersm0
@ztangent

kbarros commented:
Great idea, thanks Simon!

myersm0 commented:
hello Simon, I’ll contribute something within 1-2 weeks for my package ObservableCortex

dgleich commented:
Thanks for this! Does this suffice for GraphPlayground or do you need something more extensive? If I have time, I might be able to upload another demo that shows off mouse pointer repulsion that's fun. (Although it isn't a graph...) 

Sometimes the best way to understand data is to play around with it interactively. There is a lovely package in JavaScript to do this by Mike Bostock called d3. In particular, the d3-force library includes all sorts of fun demos. `GraphPlayground.jl` includes a port of the d3-force library to Julia to enable an interactive graph exploration tool in Makie (via `GraphMakie.jl`). It's capable of force calculations (attractions, repulsions, collisions) with tens of thousands of simple objects. I've used it to explore graphs of session conflicts at conferences as well as to generate layouts for small graphs. Given any undirected graph from `Graphs.jl`, using it is as simple as `playground(g)` to get an interactive window with labels, the ability to pin nodes, and the ability to export coordinates of the final layout.

baggepinnen commented:
Cool! Here's a snippet for Multibody
---
Multibody.jl is a library of "model components" for ModelingToolkit, an equation-based modeling language. With Multibody.jl, the user can assemble dynamical models for simulation of mechanical systems, such as robots, cars, drive lines and aircraft, by connecting ready-made components such as rods, boxes, joints and wheels. The modeling-language is responsible for deriving the equations of motion of the assembled model, lifting this burden off the shoulders of the modeler. 3D mechanical modeling benefits greatly from being able to visualize the mechanical system, both during the model-building process and while analyzing the result of a simulation. The interactive features of Makie fits this bill perfectly, allowing the user to produce both animations and interactive renders of their model.

baggepinnen commented on src/blogposts/makie-showcases/post.md:
```diff
@@ -13,6 +13,9 @@
+Multibody.jl is a library of model components for ModelingToolkit, an equation-based modeling language. With Multibody.jl, the user can assemble dynamical models for simulation of mechanical systems, such as robots, cars, drive lines and aircraft, by connecting ready-made components such as rods, boxes, joints and wheels. 
+The modeling-language is responsible for deriving the equations of motion of the assembled model, lifting this burden off the shoulders of the modeler. 3D mechanical modeling benefits greatly from being able to visualize the mechanical system, both during the model-building process and while analyzing the result of a simulation. The interactive features of Makie fits this bill perfectly, allowing ...
+
 ![](https://help.juliahub.com/multibody/dev/examples/robot.gif)
```
You may want to grab these, in case the link stops working some time in the future?

SimonDanisch replied:
They are grabbed as part of the build... But it's not perfect right now, since they get grabbed every build. I'll try to make sure it only gets downloaded once and checked into the repo

magnuask commented:
Thank you for taking the initiative! For future requests related to EnergyModelsGUI.jl, @Zetison should be tagged as he is currently the main developer for the package.

Here is a snippet for EnergyModelsGUI.jl:

EnergyModelsGUI is a graphical interface built on top of the [EnergyModelsX](https://github.com/EnergyModelsX/) framework that utilizes Makie for interactive visualizations. It provides clear, intuitive depictions of multi-energy system topologies, input data, and results, making it much easier for users to verify that their system is correctly specified and to understand model outcomes. While it does not allow altering system setups or parameters, it serves as a powerful tool for checking configurations and exploring complex results.

Larbino1 commented:
Happy for you to use VMRobotControl.jl in your blog post. This is lifted from the docs:

VMRobotControl allows you to build dynamic controllers for your robot as intuitive virtual mechanisms. You design your controller by building a virtual mechanism. You connect the virtual mechanism to the robot through compliant elements (like springs and dampers). The control action is automatically translated to the robot actuators (desired joint forces/torques).

This package is designed for building, simulating and controlling robots using passivity-based controllers represented by virtual mechanisms. This is in the tradition of virtual model control and energy shaping/damping injection. For a defined robot geometry, you can select specific coordinates and connect components to shape the robot dynamic properties. properties. This provides a simple way to program complex robot behaviour incrementally, from mechanical primitives.

ztangent commented:
This sounds like a great initiative! Thanks for featuring PDDLViz.jl. Here is a brief snippet about PDDLViz.jl and the surrounding PDDL.jl ecosystem:

```md
PDDLViz.jl is a library for visualizing, animating, and interacting with AI planning domains specified in the Planning Domain Definition Language (PDDL).
PDDLViz.jl is part of the [JuliaPlanners ecosystem](https://github.com/JuliaPlanners), and operates on top of the [PDDL.jl](https://github.com/JuliaPlanners/PDDL.jl) interpreter and compiler for PDDL.
PDDLViz.jl also supports the animation and rendering of planning algorithms and solutions supported by the [SymbolicPlanners.jl](https://github.com/JuliaPlanners/SymbolicPlanners.jl) library.
```

kbarros commented:
Thanks again! Here is our proposed blurb:

Sunny is a package for simulating atomic scale magnetism in real materials. It facilitates specification of a broad class of spin models and includes a diverse suite of numerical solvers. These include powerful methods for simulating spin dynamics both in and out of equilibrium. Uniquely, it generalizes the traditional dynamics of spin dipoles to a dynamics of SU(N) coherent states, which better captures local quantum mechanical effects. Sunny also offers a well-developed framework for calculating the dynamical spin structure factor, enabling direct comparison with scattering experiments. Ease of use is a priority, with Makie-powered GUI tools for interactive symmetry analysis and model visualization. [[arXiv:2501.13095]](https://arxiv.org/abs/2501.13095)

moyner commented:
Thanks a lot for the opportunity to add to the blogpost! Here is a blurb:

I think the image could be replaced/expanded with this one: https://sintefmath.github.io/JutulDarcy.jl/dev/assets/mostnty.BTVM5CAw.jpeg

It is a bit newer version of the same viewer and has a cooler plot in my opinion. The picture is the mesh with the porosity of a synthetic reservoir model.

> Jutul.jl is a finite-volume modelling framework based on automatic differentiation for coupled, non-linear partial differential equations that powers several different applications that include Makie extensions for visualization. This includes BattMo.jl, a package for fast and differentiable battery simulation and JutulDarcy.jl, a high-performance simulator for flow in porous media discretized on unstructured meshes.

RGerzaguet commented:
Hello,

My turn, and thanks to reach us out ! 

A small description of TempestSDR.jl 

> This package proposes a Graphical User Interface (GUI) leveraging Makie to perform real time screen eavesdrop. Screen are connected to devices by cables, often HDMI or VGA. This can lead to an unwanted emission of the image signal through electromagnetic wave. A passive listener can retrieve the image signal without be in the same room of the targeted PC!

Kevin-Mattheus-Moerman commented:
Sure. How about something like: 

[Comodo.jl](https://github.com/COMODO-research/Comodo.jl) is a Julia package for computational (bio)mechanics and computational design. Comodo offers a range of geometry processing methods. This included CAD-like operations such as extrusion, revolution, and lofting. Furthermore it offers various surface and volumetric meshing methods, which can be combined with finite element analysis. Together this provides a powerful set of tools allowing users to create frameworks for parameterised and automated design and design optimisation. Comodo is currently being extended with image-based modelling methods, such that users may segment medical image data to create surface and volumetric meshes, e.g. for finite element analysis based medical device design optimisation.

SimonDanisch commented:
Alright, I fixed a few spelling mistakes and removed the projects that haven't answered yet.
If interest exists to still publish them, we can put them in the next blogpost, which I want to release more regularly now.
I want to release this tomorrow, so please let me know until then, if anyone still wants to change something before an official release :)

myersm0 commented:
Sorry I missed the deadline for mine (ObservableCortex), but it's probably just as well because I'm working on some big changes that I would like to have finished first. Please keep it in mind for the next post.

tamasgal commented:
I will contribute next week! We (KM3NeT) have a Nature paper release next week with a GLMakie rendering as Figure 1 :)

SimonDanisch commented:
The blogpost got published here: 
https://www.linkedin.com/feed/update/urn:li:activity:7295861477326909440/
https://bsky.app/profile/makie.org/post/3li3ast7j4s2c
https://mastodon.social/@makieorg/113997848959329065
And on hackernews ;) 
Feel free to share it!
